tensorflow/core/kernels/crop_and_resize_op.cc

```diff
@@ -442,70 +455,96 @@ struct CropAndResizeBackpropImage<CPUDevice, T> {
+    // Assign ranges of boxes into different threads.
+    auto CropAndResizeBackImgPerBox = [&](int start_box, int limit_box) {
+      for (int b = start_box; b < limit_box; ++b) {
+        const float y1 = boxes(b, 0);
+        const float x1 = boxes(b, 1);
+        const float y2 = boxes(b, 2);
+        const float x2 = boxes(b, 3);
 
-      for (int y = 0; y < crop_height; ++y) {
-        const float in_y = (crop_height > 1)
-                               ? y1 * (image_height - 1) + y * height_scale
-                               : 0.5 * (y1 + y2) * (image_height - 1);
-        if (in_y < 0 || in_y > image_height - 1) {
+        const int32 b_in = box_index(b);
+        if (!FastBoundsCheck(b_in, batch_size)) {
           continue;
         }
-        const int top_y_index = floorf(in_y);
-        const int bottom_y_index = ceilf(in_y);
-        const float y_lerp = in_y - top_y_index;
 
-        for (int x = 0; x < crop_width; ++x) {
-          const float in_x = (crop_width > 1)
-                                 ? x1 * (image_width - 1) + x * width_scale
-                                 : 0.5 * (x1 + x2) * (image_width - 1);
-          if (in_x < 0 || in_x > image_width - 1) {
+        const float height_scale =
+            (crop_height > 1)
+                ? (y2 - y1) * (image_height - 1) / (crop_height - 1)
+                : 0;
+        const float width_scale =
+            (crop_width > 1) ? (x2 - x1) * (image_width - 1) / (crop_width - 1)
+                             : 0;
+
+        for (int y = 0; y < crop_height; ++y) {
+          const float in_y = (crop_height > 1)
+                                 ? y1 * (image_height - 1) + y * height_scale
+                                 : 0.5 * (y1 + y2) * (image_height - 1);
+          if (in_y < 0 || in_y > image_height - 1) {
             continue;
           }
-          if (method_name == "bilinear") {
-            const int left_x_index = floorf(in_x);
-            const int right_x_index = ceilf(in_x);
-            const float x_lerp = in_x - left_x_index;
-
-            for (int d = 0; d < depth; ++d) {
-              const float dtop = (1 - y_lerp) * grads(b, y, x, d);
-              grads_image(b_in, top_y_index, left_x_index, d) +=
-                  static_cast<T>((1 - x_lerp) * dtop);
-              grads_image(b_in, top_y_index, right_x_index, d) +=
-                  static_cast<T>(x_lerp * dtop);
-              const float dbottom = y_lerp * grads(b, y, x, d);
-              grads_image(b_in, bottom_y_index, left_x_index, d) +=
-                  static_cast<T>((1 - x_lerp) * dbottom);
-              grads_image(b_in, bottom_y_index, right_x_index, d) +=
-                  static_cast<T>(x_lerp * dbottom);
+          const int closest_y_index = roundf(in_y);
+          const int top_y_index = floorf(in_y);
+          const int bottom_y_index = ceilf(in_y);
+          const float y_lerp = in_y - top_y_index;
+          const float one_y_lerp = 1 - y_lerp;
```

Comment:
`top_y_index`, `bottom_y_index`, `y_lerp`, and `one_y_lerp` are only used in `bilinear`. Please move them into `if (isBilinear)`.

Reply:
Done.